src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = actions;
+  let clonedState = { ...state };
+
+  for (const action in actions) {
+    const point = actions[action];
+
+    if (point.type === 'addProperties') {
+      result[action] = {
```

Comment:
Instead of updating the 'result' array directly, create a new object and then push it to the 'result' array.